component/storage/indexeddb/indexeddb.go

```diff
@@ -79,18 +96,64 @@ func (p *Provider) OpenStore(name string) (storage.Store, error) {
 		return nil, err
 	}
 
-	return &store{name: name, db: p.stores[name]}, nil
+	return p.stores[name], nil
 }
 
-// SetStoreConfig is currently not implemented. It always returns a nil error so that IndexedDB can still be used as
-// a cache provider in FormatProvider for the time being.
+// SetStoreConfig sets the configuration on a store.
+// TODO (#2540): Use proper IndexedDB indexing instead of the "Tag Map" once aries-framework-go is updated to use the
+// new storage interface. The tag names used in the predefined stores (see getStoreNames()) will need to be passed in
+// somehow.
 func (p *Provider) SetStoreConfig(name string, config storage.StoreConfiguration) error {
+	store, ok := p.stores[name]
+	if !ok {
+		return storage.ErrStoreNotFound
+	}
+
+	configBytes, err := json.Marshal(config)
+	if err != nil {
+		return fmt.Errorf("failed to marshal store configuration: %w", err)
+	}
+
+	err = store.Put(storeConfigKey, configBytes)
+	if err != nil {
+		return fmt.Errorf("failed to put store store configuration: %w", err)
+	}
+
+	// Create the tag map if it doesn't exist already.
+	_, err = store.Get(tagMapKey)
+	if errors.Is(err, storage.ErrDataNotFound) {
+		err = store.Put(tagMapKey, []byte(""))
```

Comment:
Is the value here supposed to be fixed in a future PR?

Reply:
Shouldn't this be the part where the indexes are created?

Reply:
In this PR, the initial value for tagMap doesn't really matter. I just create this tagMap once so I don't have to check every time I do a Put. Subsequent Puts will update this tagMap.

This PR doesn't use the real IndexedDB indexing mechanism, which would make this code look a lot different. All I'm doing here is emulating the expected behaviour using the more basic put/get methods.